traffic-dashboard/src/pages/mocks/edit-mock-dialog/edit-mock.dialog.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Button,
+  Card,
+  CircularProgress,
+  Dialog,
+  TextField,
+  Typography,
+} from "@mui/material";
+import React, { useEffect, useState } from "react";
+import { editMock } from "../../../api/api";
+import { Mock } from "../../../types/types";
+import styles from "./edit-mock-dialog.module.scss";
+
+type EditMockDialogProps = {
+  open: boolean;
+  close: () => void;
+  mock: Omit<Mock, "active"> & { port: number };
+};
+
+type FormType = {
+  port: number;
+  method: string;
+  endpoint: string;
+  status: number;
+  data: unknown;
+};
+
+export const EditMockDialog: React.FC<EditMockDialogProps> = (props) => {
+  const { close, open, mock } = props;
+
+  const [form, setForm] = useState<FormType>({ ...mock });
+
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+
+  const { data, port, status, endpoint, method } = form;
+
+  const handleChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setForm({ ...form, [e.target.name]: e.target.value });
+  };
+
+  const handleEditMock = () => {
+    if (!port) return;
+    editMock(mock.id, port, method, endpoint, status, data)
+      .then(() => {
+        close();
+      })
+      .catch((e) => {
+        console.error(e);
+      })
+      .finally(() => {
+        setIsLoading(false);
+      });
+  };
+
+  useEffect(() => {
```

Comment:
Copying props to state is an antipattern in react

Reply:
oh, i didn't know that, thanks.